test/integration/sds_dynamic_integration_test.cc

```diff
@@ -179,29 +223,71 @@ class SdsDynamicDownstreamIntegrationTest : public SdsDynamicIntegrationBaseTest
     client_ssl_ctx_ = createClientSslTransportSocketFactory({}, context_manager_, *api_);
   }
 
+  void configToUseSds(
+      envoy::extensions::transport_sockets::tls::v3::CommonTlsContext& common_tls_context) {
+    common_tls_context.add_alpn_protocols(downstream_protocol_ == Http::CodecClient::Type::HTTP1
+                                              ? Http::Utility::AlpnNames::get().Http11
+                                              : Http::Utility::AlpnNames::get().Http3);
+
+    auto* validation_context = common_tls_context.mutable_validation_context();
+    validation_context->mutable_trusted_ca()->set_filename(
+        TestEnvironment::runfilesPath("test/config/integration/certs/cacert.pem"));
+    validation_context->add_verify_certificate_hash(TEST_CLIENT_CERT_HASH);
+
+    // Modify the listener ssl cert to use SDS from sds_cluster
+    auto* secret_config = common_tls_context.add_tls_certificate_sds_secret_configs();
+    setUpSdsConfig(secret_config, "server_cert");
+  }
+
   void createUpstreams() override {
     create_xds_upstream_ = true;
     HttpIntegrationTest::createUpstreams();
   }
 
+  void waitForSdsUpdateStats(size_t times) {
+    test_server_->waitForCounterGe(
+        listenerStatPrefix(test_quic_
+                               ? "quic_server_transport_socket_factory.context_config_update_by_sds"
+                               : "server_ssl_socket_factory.ssl_context_update_by_sds"),
```

Comment:
This makes me wonder whether the stats name should match? ie make the stat introduced in this PR `ssl_context_update_by_sds`? Or does that not fit well with the quic naming of things?

Reply:
ssl_context is a SslSocket thing. Quic transport socket factory uses its config directly instead of generating an ssl_context object. So I think use ssl_context in the stats name is confusing.